File: 백준/Silver/13305. 주유소/주유소.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const N = Number(input.shift());

const road = input.shift().split(" ").map(Number);

const gasPrice = input.shift().split(" ").map(Number);

let cost = 0;

let restDistance = road.reduce((acc, cur) => acc + cur);

for (let i = 0; i < N - 1; i++) {
  if (gasPrice[i] < gasPrice[i + 1]) {
    cost += gasPrice[i] * (road[i] + road[i + 1]);
    restDistance -= road[i] + road[i + 1];
  } else {
    cost += gasPrice[i] * road[i];
    restDistance -= road[i];
  }

  if (restDistance === 0) {
    break;
  }
}

console.log(cost);

File: 백준/Silver/13305. 주유소/주유소.py
N = int(input())
distances =list(map(int,input().split()))
prices = list(map(int,input().split()))

cost = 0
curPrice = prices[0]

for i in(range(len(distances))):
  cost += curPrice * distances[i]

  if curPrice > prices[i+1]:
    curPrice = prices[i+1]

print(cost)